internal/dataplane/translator/translate_certs.go

```diff
@@ -72,14 +79,18 @@ func (t *Translator) getGatewayCerts() []certWrapper {
 				continue
 			}
 
-			// Check if listener is marked as programmed
-			if !util.CheckCondition(
-				status.Conditions,
-				util.ConditionType(gatewayapi.ListenerConditionProgrammed),
-				util.ConditionReason(gatewayapi.ListenerReasonProgrammed),
-				metav1.ConditionTrue,
-				gateway.Generation,
-			) {
+			// Check if listener is marked as programmed when the gateway is controlled by KIC in its spec and has the "Unmanaged" annotation.
+			// If the GatewayClass is does not satify the condition, the gateway is considered to be managed by other components (for example Kong Oprator),
+			// So we do not check the "Programmed" condition before extracting the certificate from the listener.
+			if gwc.Spec.ControllerName == gatewayapi.GatewayController(t.gatewayControllerName) &&
```

Comment:
REVIEW: should we check controller name here or just check the `unmanaged` annotation?

Reply:
Already implemented controller name check here.